index.js

```diff
@@ -22,60 +22,137 @@ const copyWithTemplate = async (from, to, variables) => {
 	await writeFile(to, generatedSource);
 };
 
-const fromPath = file => path.join(__dirname, 'template', file);
-const toPath = file => path.join(process.cwd(), file);
+const getTemplateFlag = typescript => {
+	if (!typescript) {
+		return 'javascript';
+	}
+
+	return 'typescript';
+};
+
+const install = async ({typescript}) => {
+	const unpack = dependencies => {
+		return Promise.all(dependencies.map(async group => {
+			await execa('npm', ['install', ...group]);
+		}));
+	};
+
+	const jsDependencies = [
+		[
+			'ink',
+			'meow',
+			'react',
+			'prop-types',
+			'import-jsx'
+		],
+		[
+			'--save-dev',
+			'xo',
+			'ava',
+			'ink-testing-library',
+			'chalk',
+			'@babel/preset-react',
+			'@babel/register',
+			'eslint-config-xo-react',
+			'eslint-plugin-react',
+			'eslint-plugin-react-hooks'
+		]
+	];
+
+	const tsDependencies = [
+		[
+			'ink',
+			'meow',
+			'react'
+		],
+		[
+			'--save-dev',
+			'xo',
+			'ava',
+			'chalk',
+			'@babel/preset-react',
+			'@babel/register',
+			'@types/node',
+			'@types/react',
+			'@types/meow',
+			'@typescript-eslint/eslint-plugin',
+			'@typescript-eslint/parser',
+			'eslint-config-xo',
+			'eslint-config-xo-react',
+			'eslint-config-xo-typescript',
+			'eslint-plugin-react',
+			'eslint-plugin-react-hooks',
+			'ink-testing-library',
+			'ts-node',
+			'typescript',
+			'xo'
+		]
+	];
+
+	return typescript ? unpack(tsDependencies) : unpack(jsDependencies);
+};
 
-module.exports = () => {
+const copyFiles = ({typescript}) => {
+	const template = getTemplateFlag(typescript);
+	const fromPath = file => path.join(__dirname, `template/${template}`, file);
+	const toPath = file => path.join(process.cwd(), file);
 	const pkgName = slugify(path.basename(process.cwd()));
+	const variables = {
+		name: pkgName
+	};
+
+	const jsFiles = () => {
```

Comment:
Why use functions that return arrays instead of defining array variables right away?

Reply:
The methods used in the arrays are executed